crates/red_knot_python_semantic/resources/mdtest/attributes.md

```diff
@@ -2,6 +2,212 @@
 
 Tests for attribute access on various kinds of types.
 
+## Class and instance variables
+
+### Pure instance variables
+
+#### Variable only declared/defined in `__init__`
+
+Methods only defined in `__init__` are pure instance variables. They can not be accessed on the
+class itself.
+
+```py
+class C:
+    def __init__(self, value2: int) -> None:
+        self.pure_instance_variable1 = "value set in __init__"
+        self.pure_instance_variable2 = value2
+        self.pure_instance_variable3: bytes
```

Comment:
For completeness, maybe you could also have an `__init__` instance variable that is explicitly declared on assignment and is bound to a value?

```suggestion
        self.pure_instance_variable3: bytes
        self.pure_instance_variable4: bytes = b"Declared and bound"
```

I suppose it might also be good to have an instance attribute that's possible unbound as well, e.g.

```py
class C:
    def __init__(self, flag: bool):
        if flag:
            self.x: str = "oh no"
```